crates/router/src/core/payments/routing.rs

```diff
@@ -1724,15 +1741,15 @@ pub async fn perform_success_based_routing_with_open_router(
 #[instrument(skip_all)]
 pub async fn update_success_rate_score_with_open_router(
     state: &SessionState,
-    payment_connector: common_enums::RoutableConnectors,
+    payment_connector: api_routing::RoutableConnectorChoice,
     profile_id: &common_utils::id_type::ProfileId,
     payment_id: &common_utils::id_type::PaymentId,
-    payment_status: bool,
+    payment_status: common_enums::AttemptStatus,
 ) -> RoutingResult<()> {
     let open_router_req_body = or_types::UpdateScorePayload {
         merchant_id: profile_id.clone(),
-        gateway: payment_connector,
-        status: payment_status.into(),
+        gateway: payment_connector.to_string(),
+        status: routing::helpers::get_txn_status_from_attempt_status(payment_status),
```

Comment:
```suggestion
        payment_status.get_tnx_status(),
```

check if this is possible

Reply:
Yes ForeignFrom<AttemptStatus> would be better